### PR TITLE
check response status_code from authentication server

### PR DESCRIPTION
**Pull request by manelclos**

### CODE REVIEW

File: oauth2_provider/oauth2_validators.py [hunk 1]
```diff
@@ -277,6 +277,11 @@ def _get_token_from_authentication_server(
             log.exception("Introspection: Failed POST to %r in token lookup", introspection_url)
             return None
 
+        if response.status_code == 403:
```
Comment (jleclanche) [inline oauth2_provider/oauth2_validators.py]: You should check against the expected status code rather.

File: oauth2_provider/oauth2_validators.py [hunk 2]
```diff
@@ -277,6 +277,11 @@ def _get_token_from_authentication_server(
...
+            log.exception("Introspection: Failed to authenticate to %r",
```
Comment (jleclanche) [inline oauth2_provider/oauth2_validators.py]: Use log.error(). You're not in an exception.

File: oauth2_provider/oauth2_validators.py [hunk 3]
```diff
@@ -277,6 +277,11 @@ def _get_token_from_authentication_server(
...
+                          introspection_url)
```
Comment (jleclanche) [inline oauth2_provider/oauth2_validators.py]: Please don't indent like that :) And a single-line is enough here.

File: oauth2_provider/oauth2_validators.py (hunk 1 above)
Comment (manelclos) [reply oauth2_provider/oauth2_validators.py]: ok, but then we can't say that we failed to authenticate or permissions are wrong (401, 403), "if not status_code == 200", is that ok?

File: oauth2_provider/oauth2_validators.py (hunk 2 above)
Comment (manelclos) [reply oauth2_provider/oauth2_validators.py]: sure

File: oauth2_provider/oauth2_validators.py (hunk 1 above)
Comment (jleclanche) [reply oauth2_provider/oauth2_validators.py]: If you want to handle specific status codes that is fine. You should have an else clause in there though if you do.

Comment (jleclanche) [reply oauth2_provider/oauth2_validators.py]: BTW a couple of tests would be nice for this.

**Comment by manelclos** [comment]
@jleclanche, new commit ready for review. I'll squash when the review is finished.

**Pull request closed without merging**